src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -13,6 +13,11 @@
 using System.Threading;
 using System.Threading.Tasks;
 
+[assembly: UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:UnrecognizedReflectionPattern",
+    Target = "M:System.Text.Json.JsonSerializer.<<DeserializeAsyncEnumerable>g__CreateAsyncEnumerableDeserializer|27_0>d`1.MoveNext()",
```

Comment:
Appears to be based on C# generated types which in theory could change in the future

Reply:
Yes, unfortunately this is a current limitation of the trimmer with `async` methods. I'm open to other suggestions, because this is the only way I know how to suppress these warnings.

cc @vitek-karas @MichalStrehovsky @agocke

Reply:
No great answers here, unfortunately. I think the C# compiler should be involved in brainstorming a solution here. The same problem would appear if you wanted to, say, put MethodImplOptions on an async method.

Reply:
The mapping from the kickoff method to the MoveNext method is available in the [debug info](https://github.com/dotnet/corefx/blob/b8b81a66738bb10ef0790023598396861d92b2c4/src/System.Reflection.Metadata/specs/PortablePdb-Metadata.md#statemachinemethod-table-0x36) - when illinker is running in the "library mode", it could propagate the attribute (simply write it into the trimmed library). It shouldn't be too hard and it would work as a stopgap for us.

Of course this would only work if one has debug info and one runs illink in this special mode to propagate the attribute.

Reply:
> Of course this would only work if one has debug info

Which will be a problem for our customers trimming their apps that use System.Text.Json, since they won't have .pdbs for System.Text.Json at build time. So they would see these warnings.

Reply:
Yes, that's why I was suggesting we would propagate it when illink is running as part of building this repo (i.e. when it runs in the library mode, it would just look at the debug info and add a physical attribute on the MoveNext method). It would solve the problem for us right now and I don't think it's very expensive to do.